ReactWindows/ReactNative.Shared/Bridge/NativeModuleRegistry.cs

```diff
@@ -145,18 +145,19 @@ internal void NotifyReactInstanceDispose()
                 foreach (var module in _moduleInstances.Values)
                 {
                     Dispatch(module, module.OnReactInstanceDispose);
-                    module.ActionQueue?.Dispose();
+                    (module as IHasActionQueue)?.ActionQueue?.Dispose();
                 }
```

Comment:
Maybe we can just leave dispose up to the implementation of IHasActionQueue. I don't know why I ever delegated that here.

Reply:
There are a couple of subtleties with forcing the module to dispose its ActionQueue:
  - It just so happens that the default ActionQueue shouldn't be disposed by modules which conveniently frees most modules from having to dispose their ActionQueues. However, if we wanted to change the default such that each module runs on its own ActionQueue, many modules would all of a sudden have to dispose their ActionQueues.
  - The natural hook for disposing the ActionQueue would be in `OnReactInstanceDispose`. Disposing the ActionQueue here would mean disposing the ActionQueue while running on the ActionQueue. I believe `ActionQueue` handles this case correctly but it could be tricky to keep this working and for new ActionQueue implementations to get this right.

It works and I don't have any better ideas so I went forward with your suggestion.

Reply:
That's a good point. Disposing from the action queue itself is protected in a test case, but I don't know if it's something we want to do.

---
In reply to: [169539708](https://github.com/Microsoft/react-native-windows/pull/1650#discussion_r169539708) [](ancestors = 169539708)

Reply:
I added `DisposeActionQueue` to `IHasActionQueue` which resolves both of my concerns.

---
In reply to: [169584451](https://github.com/Microsoft/react-native-windows/pull/1650#discussion_r169584451) [](ancestors = 169584451,169539708)